Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -332,32 +351,74 @@ public void FailPlugin(Assembly assembly)
             ChangePluginState(plugin, PluginStatus.Malfunctioned);
         }
 
-        /// <summary>
-        /// Saves the manifest back to disk.
-        /// </summary>
-        /// <param name="manifest">The <see cref="PluginManifest"/> to save.</param>
-        /// <param name="path">The path where to save the manifest.</param>
-        /// <returns>True if successful.</returns>
+        /// <inheritdoc/>
         public bool SaveManifest(PluginManifest manifest, string path)
         {
-            if (manifest == null)
-            {
-                return false;
-            }
-
             try
             {
                 var data = JsonSerializer.Serialize(manifest, _jsonOptions);
                 File.WriteAllText(Path.Combine(path, "meta.json"), data);
```

Comment:
Can't we make `SaveManifest` async so you can serialize to a stream?
If not, `SerializeToUtf8Bytes` should be faster

Reply:
I tried to avoid refactoring since this will be backported, that seems like something that can go in a new pull request.